numbat/src/command.rs

```diff
@@ -46,49 +53,365 @@ impl FromStr for CommandKind {
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
-pub enum Command<'a> {
-    Help,
-    Info { item: &'a str },
-    List { items: Option<ListItems> },
-    Clear,
-    Save { dst: &'a str },
+#[derive(Debug, PartialEq, Eq)]
+#[must_use]
+pub enum CommandControlFlow {
+    Continue,
+    Return,
+    NotACommand,
+}
+
+pub struct CommandContext<'ctx, 'editor, Editor> {
+    pub ctx: &'ctx mut Context,
+    pub editor: &'editor mut Editor,
+}
```

Comment:
I have a minor preference to inline these two arguments instead of bundling them in a public `CommandContext` struct. Right now, they are immediately unpacked in `try_run_command` anyways. Or is there another reason for this struct?

Reply:
No reason, likely a vestige of past code.